cosmwasm/packages/std/src/imports.rs

```diff
@@ -187,6 +215,227 @@ impl Api for ExternalApi {
+    /// each of them is valid. This allows sharing computations among all signature verifications,
+    /// performing less work overall, at the cost of higher latency (the entire batch must complete),
+    /// complexity of caller code (which must assemble a batch of signatures across work-items),
+    /// and loss of the ability to easily pinpoint failing signatures.
+    ///
+    /// This batch verification implementation is adaptive, in the sense that it detects multiple
+    /// signatures created with the same verification key, and automatically coalesces terms
+    /// in the final verification equation.
+    ///
+    /// In the limiting case where all signatures in the batch are made with the same verification key,
+    /// coalesced batch verification runs twice as fast as ordinary batch verification.
+    ///
+    /// Three Variants are suppported in the input for convenience:
+    ///  - Equal number of messages, signatures, and public keys: Standard, generic functionality.
+    ///  - One message, and an equal number of signatures and public keys: Multiple digital signature
+    /// (multisig) verification of a single message.
+    ///  - One public key, and an equal number of messages and signatures: Verification of multiple
+    /// messages, all signed with the same private key.
+    ///
+    /// Any other variants of input vectors result in an error.
+    ///
+    /// Notes:
+    ///  - The "one-message, with zero signatures and zero public keys" case, is considered the empty
+    /// case.
+    ///  - The "one-public key, with zero messages and zero signatures" case, is considered the empty
+    /// case.
+    ///  - The empty case (no messages, no signatures and no public keys) returns true.
+    fn ed25519_batch_verify(
+        &self,
+        messages: &[&[u8]],
+        signatures: &[&[u8]],
+        public_keys: &[&[u8]],
+    ) -> Result<bool, VerificationError> {
+        let msgs_encoded = encode_sections(messages);
+        let msgs_send = build_region(&msgs_encoded);
+        let msgs_send_ptr = &*msgs_send as *const Region as u32;
+
+        let sigs_encoded = encode_sections(signatures);
+        let sig_sends = build_region(&sigs_encoded);
+        let sigs_send_ptr = &*sig_sends as *const Region as u32;
+
+        let pubkeys_encoded = encode_sections(public_keys);
+        let pubkeys_send = build_region(&pubkeys_encoded);
+        let pubkeys_send_ptr = &*pubkeys_send as *const Region as u32;
+
+        let result =
+            unsafe { ed25519_batch_verify(msgs_send_ptr, sigs_send_ptr, pubkeys_send_ptr) };
+        match result {
+            0 => Ok(true),
+            1 => Ok(false),
+            4 => Err(VerificationError::InvalidSignatureFormat),
+            5 => Err(VerificationError::InvalidPubkeyFormat),
+            10 => Err(VerificationError::GenericErr),
+            error_code => Err(VerificationError::unknown_err(error_code)),
```

Comment:
Feels like this should be a VerificationError::from_i32 function instead of this

Reply:
This is just a copy from CosmWasm v1